index.js

```diff
@@ -763,13 +777,19 @@ class K8sExecutor extends Executor {
             return message !== undefined;
         });
 
-        logger.info(`BuildId:${buildId}, status:${message}`);
+        logger.info(`BuildId:${buildId}, status:${status}, waitingReason:${waitingReason}, message:${message}`);
 
+        // data is not undefined when desired status or waitingReason met
+        // hence no error needs to be thrown
+        if (data !== undefined) {
```

Comment:
we want to fix this inconsistent return from the function, consider returning an obj with message and verify status and use it in worker to decide retry plus log the message